superset/db_engine_specs.py

```diff
@@ -1065,6 +1064,9 @@ def convert_to_hive_type(col_type):
                     convert_to_hive_type(column_info['type'])))
         schema_definition = ', '.join(column_name_and_type)
 
+        # Optional dependency
+        import boto3  # pylint: disable=import-error
```

Comment:
You can also keep this in the top of the file, and wrap it with an `ModuleNotFoundError`:

```python
try:
    import boto3
except ModuleNotFoundError:
    pass
```

This way you don't need the lint exception.

Reply:
I feel like it's better to raise the error `ModuleNotFoundError: No module named boto3'` than the `NameError: name 'boto3' is not defined`. The former calls the action of installing the module more clearly.

Mind if I stick with what I had?

Reply:
Ah, true! Though you could set `boto3 = None` in the block capturing the exception. I'm fine with either, it was just a nit. :)